ADALiOS/ADALiOS/ADAuthenticationContext+WebRequest.m

```diff
@@ -295,9 +296,9 @@ - (NSString*)queryStringFromResource:(NSString*)resource
     
     [startUrl appendFormat:@"&%@", [[ADLogger adalId] adURLFormEncode]];
     
-    if (![NSString adIsStringNilOrBlank:userId])
+    if (![NSString adIsStringNilOrBlank:userId.userId])
     {
-        [startUrl appendFormat:@"&%@=%@", OAUTH2_LOGIN_HINT, [userId adUrlFormEncode]];
+        [startUrl appendFormat:@"&%@=%@", OAUTH2_LOGIN_HINT, [userId.userId adUrlFormEncode]];
```

Comment:
If they gave us a UniqueId as the UserIdentifier we don't have anything else to go by, we don't have the full UserInformation struct here.